crates/router/src/services/authentication.rs

```diff
@@ -852,6 +903,7 @@ where
         let auth = AuthenticationData {
             merchant_account: merchant,
             key_store,
+            profile_id: payload.profile_id,
         };
         Ok((
             auth.clone(),
```

Comment:
I don't think .clone() is required here.

Reply:
We are using auth again in below lines.